From: Daniel Wagner <dwagner@suse.de>
Date: Thu Jun 15 11:48:53 AM PDT 2023
Subject: libtracecmd: Add explicit meson thread dependency 
Patch-mainline: mailed to mailing list
References: none
Signed-off-by: Tony Jones <tonyj@suse.de>

Older version of meson do not add automatically the pthread and dl
dependency. Thus add it explicitly to thebuild.

Signed-off-by: Daniel Wagner <dwagner@suse.de>
---
 lib/meson.build           | 3 +++
 lib/trace-cmd/meson.build | 4 ++++
 2 files changed, 7 insertions(+)

diff --git a/lib/meson.build b/lib/meson.build
index 52f892fd82d5..389281d3806a 100644
--- a/lib/meson.build
+++ b/lib/meson.build
@@ -29,6 +29,9 @@ conf = configuration_data()
 libtraceevent_dep = dependency('libtraceevent', version: '>= 1.5.0', required: true)
 libtracefs_dep = dependency('libtracefs', version: '>= 1.6.0', required: true)
 
+threads_dep = dependency('threads', required: true)
+dl_dep = cc.find_library('dl', required : false)
+
 zlib_dep = dependency('zlib', required: false)
 conf.set('HAVE_ZLIB', zlib_dep.found(), description: 'Is zlib avialable?')
 
diff --git a/lib/trace-cmd/meson.build b/lib/trace-cmd/meson.build
index 8c00ca8f8ff7..2193e5265b22 100644
--- a/lib/trace-cmd/meson.build
+++ b/lib/trace-cmd/meson.build
@@ -45,6 +45,8 @@ if libtracecmd_standalone_build
         dependencies: [
             libtraceevent_dep,
             libtracefs_dep,
+            threads_dep,
+            dl_dep,
             zlib_dep,
             libzstd_dep,
             audit_dep],
@@ -77,6 +79,8 @@ else
         dependencies: [
             libtraceevent_dep,
             libtracefs_dep,
+            threads_dep,
+            dl_dep,
             zlib_dep,
             libzstd_dep,
             audit_dep],
-- 
2.40.1

